src/core_plugins/input_control_vis/public/components/editor/index_pattern_select.js

```diff
@@ -0,0 +1,50 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import Select from 'react-select';
+
+export class IndexPatternSelect extends Component {
```

Comment:
TSVB will probably always be adhoc because there are a lot of users with dashboards that depend on the current behavior; I don't want to take that away from them.